packages/atomic/src/components/commerce/atomic-commerce-product-list/atomic-commerce-product-list.tsx

```diff
@@ -64,71 +121,204 @@ export class AtomicCommerceProductList
+      <DisplayWrapper display={this.display} listClasses={listClasses}>
+        <ResultsPlaceholdersGuard
+          density={this.density}
+          display={this.display}
+          imageSize={this.imageSize}
+          displayPlaceholders={!this.bindings.store.isAppLoaded()}
+          numberOfPlaceholders={this.productState.products.length}
+        ></ResultsPlaceholdersGuard>
+        <ItemDisplayGuard
+          firstRequestExecuted={!this.productState.isLoading}
+          hasItems={this.productState.products.length > 0}
+        >
+          {this.display === 'table'
+            ? this.renderAsTable()
+            : this.display === 'grid'
+              ? this.renderAsGrid()
+              : this.renderAsList()}
+        </ItemDisplayGuard>
+      </DisplayWrapper>
+    );
+  }
+
+  private computeListDisplayClasses() {
+    const displayPlaceholders = !this.bindings.store.isAppLoaded();
+
+    return getItemListDisplayClasses(
+      this.display,
+      this.density,
+      this.imageSize,
+
+      this.productState?.isLoading,
+      displayPlaceholders
+    );
+  }
+
+  private getPropsForAtomicProduct(product: Product) {
+    return {
+      // TODO: add back once interactive result is implemented for products
+      /* interactiveResult: buildInteractiveResult(this.bindings.engine, {
+        options: {result},
+      }), */
+      product,
+      renderingFunction: this.itemRenderingFunction,
+      loadingFlag: this.loadingFlag,
+      key: this.productListCommon.getResultId(
+        product.permanentid,
+        this.productState.responseId,
+        this.density,
+        this.imageSize
+      ),
+      content: this.productTemplateProvider.getTemplateContent(product),
+      store: this.bindings.store,
+      density: this.density,
+      imageSize: this.imageSize,
+      display: this.display,
+    };
+  }
+
+  private renderAsGrid() {
+    return this.productState.products.map((product, i) => {
+      const propsForAtomicProduct = this.getPropsForAtomicProduct(product);
+      return (
+        <DisplayGrid
+          item={{
+            ...product,
+            clickUri: product.clickUri,
+            title: product.ec_name ?? 'temp',
+          }}
+          // TODO: add back once the interactive result is implemented
+          //{...propsForAtomicProduct.interactiveResult}
+          // TODO: Remove these back once the interactive result is implemented
+          setRef={(element) =>
+            element && this.productListCommon.setNewResultRef(element, i)
+          }
+          select={function (): void {
+            throw new Error('Function not implemented.');
+          }}
+          beginDelayedSelect={function (): void {
+            throw new Error('Function not implemented.');
+          }}
+          cancelPendingSelect={function (): void {
+            throw new Error('Function not implemented.');
+          }}
```

Comment:
Added this until we add the equivalent for products. Should I instead replace DisplayGrid with a commerce-specific alternative?